### PR TITLE
#164069309 Reset user password via email

### PR DESCRIPTION
**Pull request by akram256**
**What does this PR do?**

* Resets users password via email

**Have the following endpoints working**
* api/password-reset/. This sends email to the user.
* api/password-reset/<str:token>/. retrieves the token from the link
* api/password/reset/done/. Used to update

**How should this be manually tested?**

* Pull the branch to your remote workspace
* Run the server using python manage.py runserver
* The user puts in his email into the email field by appending /api/password-reset/
On Postman add this to the body.
{
    "user": {
		  "email": "akram.mukasa@andela.com"
		}
}
* Below is a screenshot  of A verification link
screenshot 2019-03-06 at 18 17 25
screenshot 2019-03-06 at 18 18 20

* System Sends an email to the user with the password reset link upon user request
* The user receives an email with a link like http://127.0.0.1:8000/api/password-reset/eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MzEsImVtYWlsIjoiYWFjcnljZUBnbWFpbC5jb20iLCJleHAiOjE1NTM2ODg5OTl9.hqPa3zUHc2VqRaVWOi2f91VjmI36ZoMZE8-FurGQPq0/
* Clicks/copy the link on this email post it on Postman which returns user token
* Get the token and pass it on the header on Postman input method and use the following on the body'
{
  "user":{
    "password": "Akram@256"
     "confirm_password":"Akram@256"
  }
}
screenshot 2019-03-06 at 18 11 06
* if the password don't match for example
{
  "user":{
    "password": "Akram@256"
     "confirm_password":"Akram55256"
  }
}
* An error message "Password don't match" will be thrown as shown in the screenshoot below
screenshot 2019-03-06 at 18 15 39

**What are the relevant pivot tracker stores?**

#164069309

### CODE REVIEW

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -143,3 +148,33 @@ def update(self, instance, validated_data):
...
+
+
+class ResetPasswordSerializer(serializers.Serializer):
+    """
+        Handles serialization of User to reset password
+    """
+    email = serializers.EmailField(allow_blank=False)
+
+    def validate_email(self, data):
+        email = data
```
Comment (NLSanyu) [inline authors/apps/authentication/serializers.py:163]: I'm wondering why you did this:  'email = data'? It seems redundant

**Comment by PatrickCmd** [comment]
@akram256 Please do resolve the merge conflicts on this PR, move the validation module in the authentication package to the core utils module and do follow the folder structure for adding test modules as discussed earlier.

**Comment by PatrickCmd** [comment]
@akram256 Please resolve the conflicting files.

**Comment by akram256** [comment]
Hey @PatrickCmd  i have resolved my conflicts and i think am good to go

**Review by Stanley-Okwii — APPROVED** [review]
Great work 👍🏿

**Review by NLSanyu — APPROVED** [review]

**Review by PatrickCmd — APPROVED** [review]

**Pull request merged**